flor/complete_capture/transformer/transformer.py

```diff
@@ -145,9 +145,13 @@ def generic_visit(self, node):
                     if isinstance(value, ast.Assign) or \
                             isinstance(value, ast.AugAssign) or \
                             isinstance(value, ast.AnnAssign):
+                        if isinstance(value, ast.Assign) and \
+                                isinstance(value.targets[0], ast.Name) and \
+                                value.targets[0].id == '__all__' and \
+                                'dir()' in astor.to_source(value.value):
+                            value.value.generators[0].ifs = [ast.parse(astor.to_source(value.value.generators[0].ifs[0]).replace('\n', '') + "and s != 'flog'").body[0].value]
```

Comment:
The name you're iterating with: `s` won't generalize...
A better solution is to leave the first definition of __all__ as is, and then insert a statement underneath that removes flog. Here is an example:
```
__all__ = [each for each in dir()]
__all__ = [each for each in __all__ if each != 'flog]
```

Alternatively do static analysis on the list comprehension to find the name of the variable (e.g. `s`)